### PR TITLE
Added message type insted of publish and subscribe

### PR DESCRIPTION
**Pull request by vol4onok**

### CODE REVIEW

File: tests/SprykerSdkTest/AsyncApi/Console/SchemaMessageAddConsoleTest.php [hunk 1]
```diff
@@ -70,4 +70,26 @@ public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageC
         $this->assertSame(AbstractConsole::CODE_ERROR, $commandTester->getStatusCode());
         $this->assertNotEmpty($commandTester->getDisplay());
     }
+
+    /**
+     * @return void
+     */
+    public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageTypeISWrong(): void
```
Comment (stereomon) [inline tests/SprykerSdkTest/AsyncApi/Console/SchemaMessageAddConsoleTest.php:77]:
```suggestion
    public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageTypeIsNotPublishAndNotSubscribe(): void
```

File: src/SprykerSdk/AsyncApi/Console/SchemaMessageAddConsole.php [hunk 2]
```diff
@@ -124,13 +124,26 @@ public function execute(InputInterface $input, OutputInterface $output): int
...
+        }
+
         $asyncApiMessageTransfer = new AsyncApiMessageTransfer();
         $asyncApiMessageTransfer
             ->setChannel($asyncApiChannelTransfer)
             ->setName($input->getArgument(static::ARGUMENT_MESSAGE_NAME))
             ->setAddMetadata($input->getOption(static::OPTION_ADD_METADATA))
-            ->setIsPublish($input->getOption(static::OPTION_PUBLISH))
-            ->setIsSubscribe($input->getOption(static::OPTION_SUBSCRIBE));
+            ->setIsPublish($messageType === static::VALUE_PUBLISH ?: false)
```
Comment (stereomon) [inline src/SprykerSdk/AsyncApi/Console/SchemaMessageAddConsole.php:147]:
```suggestion
            ->setIsPublish($messageType === static::VALUE_PUBLISH)
```

File: src/SprykerSdk/AsyncApi/Console/SchemaMessageAddConsole.php [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 namespace SprykerSdk\AsyncApi\Console;
 
+use _PHPStan_7bd9fb728\Nette\DI\InvalidConfigurationException;
```
Comment (stereomon) [inline src/SprykerSdk/AsyncApi/Console/SchemaMessageAddConsole.php:10]: Wrong class used.

**Review by stereomon — CHANGES_REQUESTED** [review]

File: src/SprykerSdk/AsyncApi/Console/SchemaMessageAddConsole.php [hunk 4]
```diff
@@ -50,47 +51,47 @@ class SchemaMessageAddConsole extends AbstractConsole
...
      * @var string
      */
-    public const OPTION_PUBLISH = 'publish';
+    public const OPTION_MESSAGE_TYPE = 'message-type';
 
     /**
      * @var string
      */
-    public const OPTION_PUBLISH_SHORT = 'p';
+    public const OPTION_MESSAGE_TYPE_SHORT = 'e';
```
[the remark this answers is not in the document]
Comment (vol4onok) [reply src/SprykerSdk/AsyncApi/Console/SchemaMessageAddConsole.php:59]: we already have `t`

File: tests/SprykerSdkTest/AsyncApi/Console/SchemaMessageAddConsoleTest.php [hunk 5]
```diff
@@ -70,4 +73,30 @@ public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageC
...
+     * @return void
+     */
+    public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageTypeIsNotPublishAndNotSubscribe(): void
+    {
+        $commandTester = $this->tester->getConsoleTester(SchemaMessageAddConsole::class, false);
+
+        // Assert
+        $this->expectException(InvalidConfigurationException::class);
+        $this->expectExceptionMessage(sprintf(
+            'The option "%s" must either be "%s" or "%s"',
```
Comment (stereomon) [inline tests/SprykerSdkTest/AsyncApi/Console/SchemaMessageAddConsoleTest.php:87]: PLease move that to the Messages class. You used it now in two places already and with a proper method in the Messages class we keep the message itself in one place.

**Review by stereomon — APPROVED** [review]

**Pull request merged**